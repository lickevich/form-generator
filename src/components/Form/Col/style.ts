import styled from "styled-components";

export const StyledCol = styled.div`
  display: inline-block;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  &:first-child {
    width: 200px;
    padding-right: 10px;
  }
`;
