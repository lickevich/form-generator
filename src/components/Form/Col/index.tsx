import React from "react";

import { StyledCol } from "./style";

const Col: React.FC<React.ReactNode> = ({ children }) => (
  <StyledCol>{children}</StyledCol>
);

export default Col;
