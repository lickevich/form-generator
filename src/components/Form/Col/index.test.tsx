import React from "react";
import { render } from "@testing-library/react";

import Col from "./";

describe("Col", () => {
  it("render the component", () => {
    const component = (
      <Col>
        <label htmlFor="123">Label</label>
        <input type="text" id="123" />
      </Col>
    );
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
