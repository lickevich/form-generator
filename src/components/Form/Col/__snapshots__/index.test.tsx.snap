// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa iFJgCN"
  >
    <label
      for="123"
    >
      Label
    </label>
    <input
      id="123"
      type="text"
    />
  </div>
</DocumentFragment>
`;
